app/questionnaire/navigator.py

```diff
@@ -14,10 +14,12 @@ def evaluate_rule(rule, answer):
     match_value = when['value']
     condition = when['condition']
 
+    answer_to_test = answer[0] if (isinstance(answer, list) and len(answer) == 1) else answer
```

Comment:
I think this is superseded by the work in #578. Mentioning it here as it won't conflict as it's done in a different place in the other pull request.